repo/blob/storage.go

```diff
@@ -17,6 +17,18 @@ var ErrSetTimeUnsupported = errors.Errorf("SetTime is not supported")
 // ErrInvalidRange is returned when the requested blob offset or length is invalid.
 var ErrInvalidRange = errors.Errorf("invalid blob offset or length")
 
+// ErrBlobRetentionUnsupported is returned by implementations of Storage that
+// don't support setting blob-retention times.
+var ErrBlobRetentionUnsupported = errors.Errorf("setting blob-retention is not supported")
```

Comment:
do we expect callers to react to these errors programmatically in any way? if not (and they are always bubble up to the first caller and become ), they should not be public.

Reply:
Not in the context fo this PR. Having an error exposed can help with improved error response at CLI/UI. Better to keep this unexposed for now and expose it when needed on incremental PRs.